### PR TITLE
make streak only count when at least one activity is ticked, minor pe…

### PR DESCRIPTION
**Pull request by aliwoodman**
…t improvements

### CODE REVIEW

File: src/useActivities.ts [hunk 1]
```diff
@@ -68,7 +68,13 @@ export const useActivities = (
...
 
     while (allActivities[anchor.format(DATE_HASH_FORMAT)]) {
-      count++
+      const daysActivities = allActivities[anchor.format(DATE_HASH_FORMAT)]
+      if (
+        daysActivities &&
+        daysActivities.some((activity) => activity.status === Status.Done)
+      ) {
+        count++
+      }
```
Comment (jmagrippis) [inline src/useActivities.ts:77]:
```suggestion
  const dayHasDoneActivity = allActivities[anchor.format(DATE_HASH_FORMAT)].some(({ status }) => status === Status.Done)
  if (!dayHasDoneActivity) break
  count++
```

Maybe something like that? I don't mind different names, but the `while` condition means `allActivities[anchor.format(DATE_HASH_FORMAT)` will always be truthy so there is no need to check.

The `break` fixes another bug, which is the streak doesn't reset! If you skip a day it should go back to zero.

Comment (aliwoodman) [reply src/useActivities.ts:77]: Ah, good spot! Though Typescript seems to still think `allActivities[anchor.format(DATE_HASH_FORMAT)` could be undefined in the while loop somehow 
image

Comment (aliwoodman) [reply src/useActivities.ts:77]: I guess this is another example where ts can't figure out that it's defined without a variable? so options are to do what I originally did or override with `!` . Have just read a bit about this and it's not totally clear to me which is better but seems like there's not possibility the value is undefined so I'd lean towards `!` https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-0.html#non-null-assertion-operator

Comment (jmagrippis) [reply src/useActivities.ts:77]: I see! I've actually opened my mac now, and I did have something I didn't even save on the loop; but I still wasn't checking for "does it actually have any done" part.

Yes, that problem is Typescript needing a variable to understand whether something's been checking or not. It seems we can get around it by defining the variable before the while loop starts.

So maybe we end up with:

```ts
const consecutiveStreak = useMemo(() => {
    let count = 0
    const anchor = moment(currentDateHash)
    let daysActivities = allActivities[anchor.format(DATE_HASH_FORMAT)]

    while (daysActivities) {
      if (daysActivities.every((activity) => activity.status !== Status.Done)) {
        break
      }
      count++
      anchor.subtract(1, 'day')
      daysActivities = allActivities[anchor.format(DATE_HASH_FORMAT)]
    }

    return count
  }, [allActivities, currentDateHash])
```

Comment (aliwoodman) [reply src/useActivities.ts:77]: Sure!

Comment (aliwoodman) [reply src/useActivities.ts:77]: (will commit and merge)

**Pull request merged**